services/management/dbaas/psmdb_cluster_service.go

```diff
@@ -169,17 +173,31 @@ func (s PSMDBClusterService) UpdatePSMDBCluster(ctx context.Context, req *dbaasv
 			Kubeconfig: kubernetesCluster.KubeConfig,
 		},
 		Name: req.Name,
-		Params: &dbaascontrollerv1beta1.UpdatePSMDBClusterRequest_UpdatePSMDBClusterParams{
-			ClusterSize: req.Params.ClusterSize,
-			Replicaset: &dbaascontrollerv1beta1.UpdatePSMDBClusterRequest_UpdatePSMDBClusterParams_ReplicaSet{
-				ComputeResources: &dbaascontrollerv1beta1.ComputeResources{
-					CpuM:        req.Params.Replicaset.ComputeResources.CpuM,
-					MemoryBytes: req.Params.Replicaset.ComputeResources.MemoryBytes,
-				},
-			},
-		},
 	}
 
+	if req.Params.Suspend && req.Params.Resume {
+		return nil, status.Error(codes.InvalidArgument, "resume and suspend cannot be set together")
+	}
+
+	if req.Params != nil {
+		in.Params = &dbaascontrollerv1beta1.UpdatePSMDBClusterRequest_UpdatePSMDBClusterParams{
+			ClusterSize: req.Params.ClusterSize,
+		}
+
+		in.Params.Suspend = req.Params.Suspend
+		in.Params.Resume = req.Params.Resume
+
+		if req.Params.Replicaset != nil {
+			in.Params.Replicaset = new(dbaascontrollerv1beta1.UpdatePSMDBClusterRequest_UpdatePSMDBClusterParams_ReplicaSet)
+		}
+
+		if req.Params.Replicaset.ComputeResources != nil {
```

Comment:
it might raise an error if Replicaset is nil

Reply:
Yes, correct. Fixed.